pkg/controller/elasticsearch/user/filerealm/role_users.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package filerealm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/elastic/cloud-on-k8s/pkg/utils/set"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/stringsutil"
+)
+
+// roleUsersMapping is a map {role name -> [] user names}
+type roleUsersMapping map[string][]string
```

Comment:
There is a concept of role mapping in Elasticsearch (which you don't mean here) but the naming choices are  somewhat confusing close to that here https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-roles.html

Reply:
ah indeed! I should rename it.
`usersRoles` would be closer to the `users_roles` file, but I don't like how it seems to express a {user -> roles} relationship whereas we actually have {role -> users}.
Any naming suggestion?

Reply:
Not really, your original name was actually good. I see your point about the inverse logic of the `users_roles` naming, but maybe it's still worth considering given that it would be recognisable to readers who are familiar with the files generated.

Reply:
Let's do `usersRoles` but make sure we explain the reverse mapping in comments.